src/main/java/duke/Ui.java

```diff
@@ -0,0 +1,117 @@
+        System.out.println(str);
+        System.out.println("-----");
+    }
+
+    public void Greet() {
+        dukePrintLn(GREETING);
+    }
+
+    public void readInput() {
+
+        try {
+            String nextCommand = sc.nextLine();
+            while (!nextCommand.equals(CommandsEnum.bye.toString())) {
+                executeCommand(nextCommand);
+                nextCommand = sc.nextLine();
+            }
+            dukePrintLn(BYE);
+        } catch (UnknownCommandException | NumberFormatException e) {
+            System.out.println("Error: " + e.getMessage());
+            readInput();
+        }
+    }
+
+    public void executeCommand(String nextCommand) throws UnknownCommandException {
+        if (nextCommand.equals(CommandsEnum.list.toString())) {
+            taskList.displayAllTask();
+        } else if (nextCommand.startsWith(CommandsEnum.mark.toString())) {
+            int index = convertToInt(nextCommand);
+            storage.toggleDone(index, true);
+            taskList.setTaskAsDone(index);
+        } else if (nextCommand.startsWith(CommandsEnum.unmark.toString())) {
+            int index = convertToInt(nextCommand);
+            storage.toggleDone(index, false);
+            taskList.setTaskAsUndone(index);
+        } else if (nextCommand.contains(CommandsEnum.todo.toString()) || nextCommand.contains(CommandsEnum.deadline.toString()) || nextCommand.contains(CommandsEnum.event.toString())) {
+            taskList.addTask(nextCommand);
+            storage.addLineToFile(nextCommand);
+
+        } else if (nextCommand.startsWith(CommandsEnum.delete.toString())) {
+            int index = convertToInt(nextCommand);
+            storage.deleteLine(index);
+            taskList.deleteTask(index);
+        } else if(nextCommand.startsWith(CommandsEnum.find.toString())){
+            String name = Parser.getFindObject(nextCommand);
+            taskList.findTask(name);
+        }
+        else {
+            throw new UnknownCommandException();
+        }
+
+
+    }
+
+
+    public String getResponse(String nextCommand) throws UnknownCommandException {
+        if(nextCommand.equals(CommandsEnum.bye.toString())){
+            return BYE;
+        }
+        else if (nextCommand.equals(CommandsEnum.list.toString())) {
+            return taskList.displayAllTask();
+        } else if (nextCommand.startsWith(CommandsEnum.mark.toString())) {
+            int index = convertToInt(nextCommand);
+            storage.toggleDone(index, true);
+            return taskList.setTaskAsDone(index);
+        } else if (nextCommand.startsWith(CommandsEnum.unmark.toString())) {
+            int index = convertToInt(nextCommand);
+            storage.toggleDone(index, false);
+            return taskList.setTaskAsUndone(index);
+        } else if (nextCommand.contains(CommandsEnum.todo.toString()) || nextCommand.contains(CommandsEnum.deadline.toString()) || nextCommand.contains(CommandsEnum.event.toString())) {
```

Comment:
I noticed the condition in the `else if` statement has exceeded the hard limit of 120 characters. I have a few suggestions, firstly to use line wrap at appropriate place, abstract out the condition in a Boolean variable or abstracting this big method into smaller methods. I noticed this issue in other `else if` statements in this methods.